src/nwb/NWBFile.hpp

```diff
@@ -46,12 +46,12 @@ class NWBFile
    * @brief Initializes the NWB file by opening and setting up the file
    * structure.
    */
-  void initialize();
+  Status initialize();
```

Comment:
Please add a comment to the docs of all functions that are not accessible when the file is in recording mode, i.e., `initialize`, `finalize` etc.. Maybe something along the lines of: "If the file is in recording mode (i.e., when isRecording is true), then this function will fail. To modify the structure of the NWB file (i.e., add datasets, groups, etc.) the recording must be paused or stopped)"

Reply:
Added this text: "Note, this function will fail if the file is in a mode where new objects cannot be added"